token-metadata/program/src/utils.rs

```diff
@@ -55,79 +55,81 @@ pub fn assert_data_valid(
     if data.seller_fee_basis_points > 10000 {
         return Err(MetadataError::InvalidBasisPoints.into());
     }
-    if data.creators.is_some() {
-        if let Some(creators) = &data.creators {
-            if creators.len() > MAX_CREATOR_LIMIT {
-                return Err(MetadataError::CreatorsTooLong.into());
-            }
 
-            if creators.is_empty() {
-                return Err(MetadataError::CreatorsMustBeAtleastOne.into());
-            } else {
-                let mut found = false;
-                let mut total: u8 = 0;
-                for i in 0..creators.len() {
-                    let creator = &creators[i];
-                    for iter in creators.iter().skip(i + 1) {
-                        if iter.address == creator.address {
-                            return Err(MetadataError::DuplicateCreatorAddress.into());
-                        }
-                    }
+    if let Some(creators) = &data.creators {
+        if creators.len() > MAX_CREATOR_LIMIT {
+            return Err(MetadataError::CreatorsTooLong.into());
+        }
 
-                    total = total
-                        .checked_add(creator.share)
-                        .ok_or(MetadataError::NumericalOverflowError)?;
+        if creators.is_empty() {
+            return Err(MetadataError::CreatorsMustBeAtleastOne.into());
+        }
 
-                    if creator.address == *update_authority {
-                        found = true;
-                    }
+        // Do not allow duplicate creators.
+        let u: Vec<&Creator> = creators.into_iter().unique().collect();
```

Comment:
This check won't work as-is because it would allow for creators with different struct member fields to count as unique entries when it really needs to be comparing on `creator.address`.